lib/phenotype/versioned_app.rb

```diff
@@ -0,0 +1,48 @@
+module Phenotype
+  class VersionedApp
+    attr_accessor :strategies, :routes
+    attr_writer :no_strategy_response, :too_many_strategies_response, :version_not_found_response
+
+    def initialize(strategies: [])
+      @routes = {}
+      @strategies = strategies
+    end
+
+    def add_strategy(strat)
+      strategies.push(strat)
+    end
+
+    def add_version(version:, &block)
+      routes[version] = block
+    end
+
+    def call(env)
+      strats = strategies.find_all { |strat| strat.version(env) }
+      return no_strategy_response.call(env) if strategies.empty?
```

Comment:
A potential solution would be a separate middleware that checks for the presence of a version and adds it if not found. That keeps this library nice and clean and doesn't suggest to future users that providing a default is something that should normally be done.

@wshatch is out until tomorrow, but I'd like to wait to get his thoughts on this.